spec/lib/heartcheck/executors/hash_response_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'heartcheck/executors/hash_response'
+
+module Heartcheck
+    module Executors
+      describe HashResponse do
+        subject { described_class.new }
+  
+        describe '#dispatch' do
+          let(:registered) { 3 }
+  
+          let(:checkers) do
+            registered.times.collect do |current|
+              double.tap do |checker|
```

Comment:
What about using `instance_double` to avoid stubbing non-existing methods?

I think you can avoid using the method `tap` and let the code more readable, like this:

```ruby
registered.times.collect do |current|
  instance_double(
    Heartcheck::Checks::Base,
    check: { "check_#{current}".to_sym => { status: 'ok' } }
  )
end
```